### PR TITLE
Adding support to set query job priority

### PR DESCRIPTION
**Pull request by suryasoma**

### CODE REVIEW

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java [hunk 1]
```diff
@@ -348,7 +350,10 @@ Job create(JobInfo jobInfo) {
 
   public TableResult query(String sql) {
     try {
-      return bigQuery.query(QueryJobConfiguration.of(sql));
```
Comment (davidrabinowitz) [inline bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java:351]: please convert to use `JobConfigurationFactory.createQueryJobConfigurationBuilder()`

File: spark-bigquery-dsv2/spark-3.3-bigquery/src/test/java/com/google/cloud/spark/bigquery/integration/Spark33ReadFromQueryIntegrationTest.java [hunk 2]
```diff
@@ -18,5 +18,4 @@
 public class Spark33ReadFromQueryIntegrationTest extends ReadFromQueryIntegrationTestBase {
 
   // tests are from the super-class
-
```
Comment (davidrabinowitz) [inline spark-bigquery-dsv2/spark-3.3-bigquery/src/test/java/com/google/cloud/spark/bigquery/integration/Spark33ReadFromQueryIntegrationTest.java:21]: please revert the redundant change

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: spark-bigquery-dsv2/spark-3.3-bigquery/src/test/java/com/google/cloud/spark/bigquery/integration/Spark33ReadFromQueryIntegrationTest.java (hunk 2 above)
Comment (suryasoma) [reply spark-bigquery-dsv2/spark-3.3-bigquery/src/test/java/com/google/cloud/spark/bigquery/integration/Spark33ReadFromQueryIntegrationTest.java]: Done

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java (hunk 1 above)
Comment (suryasoma) [reply bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java:351]: Ack.

**Review by davidrabinowitz — APPROVED** [review]

**Pull request merged**